pyptr.h

```diff
@@ -0,0 +1,46 @@
+#ifndef PYPTR_H
+#define PYPTR_H
+#pragma once
+
+#include <Python.h>
+
+/**
+ * Implements a smart pointer to a PyObject.
+ */
+template <class O = PyObject>
+class PyPtr {
+public:
+    // assumes a new reference; use "borrowed" otherwise
+    PyPtr(O* o) : _obj(o) {}
+
+    PyPtr(const PyPtr& ptr) : _obj(ptr._obj) {
+        Py_IncRef((PyObject*)_obj);
+    }
+
+    static PyPtr borrowed(O* o) {
+        Py_IncRef((PyObject*)o);
+        return PyPtr(o);
+    }
+
+    O* operator->() { return _obj; }
+
+    operator O*() { return _obj; }
+
+
+    PyPtr& operator=(const PyPtr& ptr) {
```

Comment:
This is stale - the Scalene version does an equality test, and I think has more comments as well.

Reply:
I merged the scalene changes.